ex5B/ex5.js

```diff
@@ -10,9 +10,46 @@
  * firstUniqueChar("aaaaaaaaa") === null
  * firstUniqueChar("a") === "a"
  */
-
 function firstUniqueChar(str) {
 
+        var charList = {}, firstUniqueCharVal = null;
+
+        for (var searchIndex = 0; searchIndex < str.length; searchIndex++) {
+
+                if (charList[str.charAt(searchIndex)] === undefined) {
+
+			// Creat new property
+			charList[str.charAt(searchIndex)] = 1;
+
+                } else {
+			// Add to property
+                        charList[str.charAt(searchIndex)] += 1;
+
+                }
+        }
+
+        for (searchIndex = 0; searchIndex < Object.keys(charList).length; searchIndex++) {
```

Comment:
You don't know how much work `Object.keys(charList)` require. Please cache this value into a variable (it will also read better).